src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Common/Builder.cs

```diff
@@ -393,12 +393,36 @@ protected bool BuildInternal (string projectOrSolution, string target, string []
 			}
 			if (!result && ThrowOnBuildFailure) {
 				string message = "Build failure: " + Path.GetFileName (projectOrSolution) + (BuildLogFile != null && File.Exists (buildLogFullPath) ? "Build log recorded at " + buildLogFullPath : null);
-				throw new FailedBuildException (message, null, File.ReadAllText (buildLogFullPath));
+				//NOTE: enormous logs will lock up IDE's UI
+				if (IsRunningInIDE) {
+					throw new FailedBuildException (message);
```

Comment:
Should the exception message include `buildLogFullPath`, so that you know which log file to read?  Or would this be "obvious" in normal use cases?

Reply:
`message` will say `$"Build failure: {projectOrSolution} Build log recorded at {buildLogFullPath}"`.

That second argument on `FailedBuildException`'s ctor is `innerException`, which seems to always get passed a `null`.